x-pack/plugins/maps/public/actions/layer_actions.ts

```diff
@@ -724,3 +721,21 @@ export function updateMetaFromTiles(layerId: string, mbMetaFeatures: TileMetaFea
     await dispatch(updateStyleMeta(layerId));
   };
 }
+
+function clearInspectorAdapters(layer: ILayer, adapters: Adapters) {
+  if (!layer.getSource().isESSource()) {
+    return;
+  }
+
+  if (adapters.vectorTiles) {
+    adapters.vectorTiles.removeLayer(layer.getId());
+  }
+
+  if (adapters.requests && 'getValidJoins' in layer) {
+    const vectorLayer = layer as IVectorLayer;
+    adapters.requests!.resetRequest((layer.getSource() as IESSource).getId());
```

Comment:
I don't understand why we need the non-null assertions for `adapters.requests` when the if statement already checks if `adapters.requests` exists? What am I missing?

Reply:
Not sure either. tslint was throwing as error. I am not sure tslint recognized check above.